### PR TITLE
 Add function rcutils_string_array_cmp

### PR DESCRIPTION
**Pull request by jacobperron**
Useful for comparing two string arrays according to lexographical order.

### CODE REVIEW

File: src/string_array.c [hunk 1]
```diff
@@ -87,6 +88,53 @@ rcutils_string_array_fini(rcutils_string_array_t * string_array)
...
+    return RCUTILS_RET_INVALID_ARGUMENT;
+  }
+  if (NULL == lhs->data || NULL == rhs->data) {
+    RCUTILS_SET_ERROR_MSG("string_array data is null");
+    return RCUTILS_RET_INVALID_ARGUMENT;
+  }
+  if (NULL == res) {
+    RCUTILS_SET_ERROR_MSG("res argument is null");
+    return RCUTILS_RET_INVALID_ARGUMENT;
+  }
```
Comment (hidmic) [inline src/string_array.c]: @jacobperron for above checks, why not use `RCUTILS_CHECK_ARGUMENT_FOR_NULL()` and `RCUTILS_CHECK_FOR_NULL_WITH_MSG()` macros instead?

File: include/rcutils/types/string_array.h [hunk 2]
```diff
@@ -110,6 +110,27 @@ RCUTILS_WARN_UNUSED
...
+/**
+ * The two string arrays are compared according to lexographical order.
+ *
+ * \param[in] sa0 The first string array.
+ * \param[in] sa1 The second string array.
+ * \param[out] res Negative value if `lhs` appears before `rhs` in lexographical order.
+ *   Zero if `lhs` and `rhs` are equal.
+ *   Positive value if `lhs` appears after `rhs in lexographical order.
+ * \return `RCUTILS_RET_OK` if successful, or
+ * \return `RCUTILS_RET_INVALID_ARGUMENT` if there is an invalid arguments, or
```
Comment (hidmic) [inline include/rcutils/types/string_array.h]: @jacobperron consider describing what an invalid argument is in this context.

File: src/string_array.c (hunk 1 above)
Comment (jacobperron) [reply src/string_array.c]: I was looking at how the error handling was done in the other functions in this file. Updated this function in 5774096. We can update the others in a follow-up PR.

File: include/rcutils/types/string_array.h (hunk 2 above)
Comment (jacobperron) [reply include/rcutils/types/string_array.h]: Done 2962266

**Comment by jacobperron** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/6639/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/2965/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/5451/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/6464/)

**Comment by jacobperron** [comment]
This feature is no longer needed for ros2/ros2#677.

**Pull request merged**